tests/rptest/services/storage.py

```diff
@@ -135,6 +135,14 @@ def partitions(self, ns, topic):
                 return [p[1] for p in parts.items()]
         return []
 
+    def segments(self, ns: str, topic: str,
+                 partition_idx: int) -> list[Segment]:
+        partitions = self.partitions(ns, topic)
+        if len(partitions) <= partition_idx:
+            return []
```

Comment:
do you think this should return `None`? there seems to be a useful distinction between "the partition you requested is unknown" and zero segments.

Reply:
Think it would make sense. It's a good point.

Reply:
Changed it to throw an exception. If a test cares about this state it can catch it.